File: lib/core/constants/user_settings_keys.dart
/// 全局设置持久化键常量，统一跨 Flutter 与原生的访问命名。
class UserSettingKeys {
  UserSettingKeys._();

  // 显示与列表
  static const String displayMode = 'display_mode';
  static const String sortMode = 'sort_mode';
  static const String privacyModeEnabled = 'privacy_mode_enabled';
  static const String bottomNavigationItems = 'bottom_navigation_items';

  // 截屏基础配置
  static const String screenshotInterval = 'screenshot_interval';
  static const String screenshotEnabled = 'screenshot_enabled';
  static const String autoAddNewAppsToCapture = 'auto_add_new_apps_to_capture';
  static const String windowScreenshotApiEnabled =
      'window_screenshot_api_enabled';
  static const String screenshotDedupeMode = 'screenshot_dedupe_mode';
  static const String imageFormat = 'image_format';
  static const String imageQuality = 'image_quality';
  static const String useTargetSize = 'use_target_size';
  static const String targetSizeKb = 'target_size_kb';
  static const String aiImageSendFormat = 'ai_image_send_format';
  static const String screenshotExpireEnabled = 'screenshot_expire_enabled';
  static const String screenshotExpireDays = 'screenshot_expire_days';
  static const String aiRawResponseCleanupEnabled =
      'ai_raw_response_cleanup_enabled';
  static const String aiRawResponseCleanupDays = 'ai_raw_response_cleanup_days';
  static const String aiRawResponseCleanupLastTs =
      'ai_raw_response_cleanup_last_ts';

  // 时间段总结（Segment）与 AI 请求限制
  static const String segmentSampleIntervalSec = 'segment_sample_interval_sec';
  static const String segmentDurationSec = 'segment_duration_sec';
  static const String aiMinRequestIntervalSec = 'ai_min_request_interval_sec';
  static const String dynamicAutoRepairEnabled = 'dynamic_auto_repair_enabled';
  static const String dynamicRebuildDayConcurrency =
      'dynamic_rebuild_day_concurrency';
  // 动态合并（仅自动合并；强制合并不受限）
  static const String mergeDynamicMaxSpanSec = 'merge_dynamic_max_span_sec';
  static const String mergeDynamicMaxGapSec = 'merge_dynamic_max_gap_sec';

  // 通知提醒
  static const String dailyNotifyEnabled = 'daily_notify_enabled';
  static const String dailyNotifyHour = 'daily_notify_hour';
  static const String dailyNotifyMinute = 'daily_notify_minute';
  static const String morningNotifyEnabled = 'morning_notify_enabled';

  // 动态页 UI
  static const String dynamicEntryLogIconEnabled =
      'dynamic_entry_log_icon_enabled';

  // 日志管理
  static const String logRetentionDays = 'log_retention_days';

  // 百度网盘自动云备份
  static const String cloudBackupEnabled = 'cloud_backup_enabled';
  static const String cloudBackupFrequencyDays = 'cloud_backup_frequency_days';
  static const String cloudBackupAllowMobileData =
      'cloud_backup_allow_mobile_data';
  static const String cloudBackupKeepLatestCount =
      'cloud_backup_keep_latest_count';
  static const String cloudBackupBaiduAppKey = 'cloud_backup_baidu_app_key';
  static const String cloudBackupBaiduSecretKey =
      'cloud_backup_baidu_secret_key';
  static const String cloudBackupBaiduAuthorizationCode =
      'cloud_backup_baidu_authorization_code';
  static const String cloudBackupBaiduAccessToken =
      'cloud_backup_baidu_access_token';
  static const String cloudBackupBaiduRefreshToken =
      'cloud_backup_baidu_refresh_token';
  static const String cloudBackupBaiduTokenExpiresAt =
      'cloud_backup_baidu_token_expires_at';
  static const String cloudBackupLastSuccessAt = 'cloud_backup_last_success_at';
  static const String cloudBackupLastAttemptAt = 'cloud_backup_last_attempt_at';
  static const String cloudBackupLastStatus = 'cloud_backup_last_status';
  static const String cloudBackupDeviceId = 'cloud_backup_device_id';
  static const String cloudBackupProgressStage = 'cloud_backup_progress_stage';
  static const String cloudBackupProgressPercent =
      'cloud_backup_progress_percent';
  static const String cloudBackupProgressDetail =
      'cloud_backup_progress_detail';
  static const String cloudBackupProgressUpdatedAt =
      'cloud_backup_progress_updated_at';
  static const String cloudBackupProgressBytesDone =
      'cloud_backup_progress_bytes_done';
  static const String cloudBackupProgressBytesTotal =
      'cloud_backup_progress_bytes_total';
  static const String cloudBackupProgressActive =
      'cloud_backup_progress_active';

  // 自动更新
  static const String autoUpdateEnabled = 'auto_update_enabled';
  static const String autoUpdateLastCheckMs = 'auto_update_last_check_ms';
  static const String autoUpdateIgnoredVersion = 'auto_update_ignored_version';
}

/// 兼容旧版 SharedPreferences 中的键名，用于迁移历史数据。
class LegacySettingKeys {
  LegacySettingKeys._();

  static const List<String> screenshotInterval = <String>[
    'timed_screenshot_interval',
    'flutter.screenshot_interval',
  ];
}

File: android/app/src/main/kotlin/com/fqyw/screen_memo/settings/UserSettingsKeys.kt
package com.fqyw.screen_memo.settings
object UserSettingsKeysNative {
    const val DISPLAY_MODE = "display_mode"
    const val SORT_MODE = "sort_mode"
    const val PRIVACY_MODE_ENABLED = "privacy_mode_enabled"
    const val SCREENSHOT_INTERVAL = "screenshot_interval"
    const val SCREENSHOT_ENABLED = "screenshot_enabled"
    const val AUTO_ADD_NEW_APPS_TO_CAPTURE = "auto_add_new_apps_to_capture"
    const val WINDOW_SCREENSHOT_API_ENABLED = "window_screenshot_api_enabled"
    const val SCREENSHOT_DEDUPE_MODE = "screenshot_dedupe_mode"
    const val IMAGE_FORMAT = "image_format"
    const val IMAGE_QUALITY = "image_quality"
    const val USE_TARGET_SIZE = "use_target_size"
    const val TARGET_SIZE_KB = "target_size_kb"
    const val AI_IMAGE_SEND_FORMAT = "ai_image_send_format"
    const val SCREENSHOT_EXPIRE_ENABLED = "screenshot_expire_enabled"
    const val SCREENSHOT_EXPIRE_DAYS = "screenshot_expire_days"
    const val SEGMENT_SAMPLE_INTERVAL_SEC = "segment_sample_interval_sec"
    const val SEGMENT_DURATION_SEC = "segment_duration_sec"
    const val AI_MIN_REQUEST_INTERVAL_SEC = "ai_min_request_interval_sec"
    const val DYNAMIC_AUTO_REPAIR_ENABLED = "dynamic_auto_repair_enabled"
    const val DYNAMIC_REBUILD_DAY_CONCURRENCY = "dynamic_rebuild_day_concurrency"
    // 动态合并（仅自动合并；强制合并不受限）
    const val MERGE_DYNAMIC_MAX_SPAN_SEC = "merge_dynamic_max_span_sec"
    const val MERGE_DYNAMIC_MAX_GAP_SEC = "merge_dynamic_max_gap_sec"
    const val MERGE_DYNAMIC_MAX_IMAGES = "merge_dynamic_max_images"
    const val DAILY_NOTIFY_ENABLED = "daily_notify_enabled"
    const val DAILY_NOTIFY_HOUR = "daily_notify_hour"
    const val DAILY_NOTIFY_MINUTE = "daily_notify_minute"
    const val CLOUD_BACKUP_ENABLED = "cloud_backup_enabled"
    const val CLOUD_BACKUP_FREQUENCY_DAYS = "cloud_backup_frequency_days"
    const val CLOUD_BACKUP_ALLOW_MOBILE_DATA = "cloud_backup_allow_mobile_data"
    const val CLOUD_BACKUP_KEEP_LATEST_COUNT = "cloud_backup_keep_latest_count"
    const val CLOUD_BACKUP_BAIDU_APP_KEY = "cloud_backup_baidu_app_key"
    const val CLOUD_BACKUP_BAIDU_SECRET_KEY = "cloud_backup_baidu_secret_key"
    const val CLOUD_BACKUP_BAIDU_AUTHORIZATION_CODE = "cloud_backup_baidu_authorization_code"
    const val CLOUD_BACKUP_BAIDU_ACCESS_TOKEN = "cloud_backup_baidu_access_token"
    const val CLOUD_BACKUP_BAIDU_REFRESH_TOKEN = "cloud_backup_baidu_refresh_token"
    const val CLOUD_BACKUP_BAIDU_TOKEN_EXPIRES_AT = "cloud_backup_baidu_token_expires_at"
    const val CLOUD_BACKUP_LAST_SUCCESS_AT = "cloud_backup_last_success_at"
    const val CLOUD_BACKUP_LAST_ATTEMPT_AT = "cloud_backup_last_attempt_at"
    const val CLOUD_BACKUP_LAST_STATUS = "cloud_backup_last_status"
    const val CLOUD_BACKUP_DEVICE_ID = "cloud_backup_device_id"
    const val CLOUD_BACKUP_PROGRESS_STAGE = "cloud_backup_progress_stage"
    const val CLOUD_BACKUP_PROGRESS_PERCENT = "cloud_backup_progress_percent"
    const val CLOUD_BACKUP_PROGRESS_DETAIL = "cloud_backup_progress_detail"
    const val CLOUD_BACKUP_PROGRESS_UPDATED_AT = "cloud_backup_progress_updated_at"
    const val CLOUD_BACKUP_PROGRESS_BYTES_DONE = "cloud_backup_progress_bytes_done"
    const val CLOUD_BACKUP_PROGRESS_BYTES_TOTAL = "cloud_backup_progress_bytes_total"
    const val CLOUD_BACKUP_PROGRESS_ACTIVE = "cloud_backup_progress_active"
}

object LegacySettingKeysNative {
    val SCREENSHOT_INTERVAL = listOf(
        "timed_screenshot_interval",
        "flutter.screenshot_interval"
    )
}


